base/math.jl

```diff
@@ -999,6 +999,7 @@ end
 @constprop :aggressive function ^(x::Float64, y::Float64)
     yint = unsafe_trunc(Int, y) # Note, this is actually safe since julia freezes the result
     y == yint && return x^yint
+    x === -1.0 && y >= 2*inv(eps()) && return 1.0
```

Comment:
I think the regression from #44452 happens for more than just specifically `-1.0`:

```julia
julia> VERSION
v"1.7.0"

julia> (-0.999999)^(1.0e63)
0.0

```

```julia
julia> VERSION
v"1.9.0-DEV.135"

julia> (-0.999999)^(1.0e63)
ERROR: DomainError with -0.999999:
Exponentiation yielding a complex result requires a complex argument.
Replace x^y with (x+0im)^y, Complex(x)^y, or similar.
Stacktrace:
 [1] throw_exp_domainerror(x::Float64)
   @ Base.Math ./math.jl:37
 [2] ^(x::Float64, y::Float64)
   @ Base.Math ./math.jl:1002
 [3] top-level scope
   @ REPL[60]:1
```

Reply:
yuck. This is moderately annoying.

Reply:
should be fixed.

Reply:
LGTM, maybe could use a couple more tests